apps/webapp/app/send/send-form.tsx

```diff
@@ -15,27 +19,49 @@ const Send = () => {
     amount,
     asset,
     recipient,
-    memo,
-    hidden,
+    memoText,
+    //memoSender,
     setAmount,
     setAsset,
     setRecipient,
-    setMemo,
-    setHidden,
+    setMemoText,
+    setMemoSender,
     sendTx,
     txInProgress,
   } = useStore(sendSelector);
 
+  const [hidden, setHidden] = useState(false);
   const assetBalance = useSendBalance();
   const validationErrors = sendValidationErrors(asset, amount, recipient, assetBalance);
 
+  // TODO: enable address selection
+  const defaultSender = useAddress(0);
+  const ephemeralSender = useEphemeralAddress(0, { enabled: hidden });
+
+  const sender = useMemo(() => {
+    if (!hidden && defaultSender.data) return bech32Address(defaultSender.data);
+    if (hidden && ephemeralSender.data) return bech32Address(ephemeralSender.data);
+    return '';
+  }, [hidden, defaultSender, ephemeralSender]);
+
+  useEffect(() => {
+    setMemoSender(sender);
+  }, [sender, setMemoSender]);
+
```

Comment:
when i wrote it this way i was thinking:

* avoid generating the ephemeral address until it's required, to minimize requests. it looks like this one is local and might be computationally simple.
* ephemeral addresses should not be reused. if we pull an address from the store, how do we ensure the address is new and specific to this transaction? we're just going to have to trigger a request.
* we should show the actual selected address to the user. there's a conceptual symmetry here, as the user must provide a destination address. in the case of a non-ephemeral sender, it is additionally confirming and helps the user understand "i'm transmitting *my* address" in a correlateable way

if we don't display the address to the user, the points are all eliminated and it's fine to just set the page controls into the store like before, putting off the requests to sendTx. in discussion i think we had talked specifically about showing the address, so that's the approach i took.

Reply:
== Update ==
A good deal of refactoring of data fetching has just been merged. We essentially have two paths for retrieving data:
- Loaders (preferred)
  - Fetches on route changes and does not load the component until data is available. We now have a bunch of places in the code base where this is used.
  - Sometimes it may seem like this is not an option, but we often have more opportunities to change/add to the URL path than we realize.
- In-component useQuery hooks
  - If changing the route (in response to user actions) is not really possible or the right flow, we can always just "reactively" fetch using react-query. However, we should be careful not to fall into the same pit we had before where there were like +5 hooks composing with each other. If we go this route, let's try our best to use a single useQuery hook and compose within the async fetch function. 
  - In this PR's case, I think we likely are in this bucket